### PR TITLE
refactor(web): use queries to track progress

### PR DESCRIPTION
**Pull request by imobachgs**
Follow-up of #1439 the goal of this PR is to change the progress track to use Tanstack Query.

### CODE REVIEW

File: web/src/queries/progress.ts [hunk 1]
```diff
@@ -0,0 +1,91 @@
...
+    queryKey: ["progress", service],
+    queryFn: () =>
+      fetch(`/api/${service}/progress`)
+        .then((res) => res.json())
+        .then((body) => Progress.fromApi(body)),
+  };
+};
+
+type UseProgressOptions = {
+  suspense: boolean;
```
Comment (dgdavid) [inline web/src/queries/progress.ts:44]: NP: should `suspense` be optional?

File: web/src/queries/progress.ts [hunk 2]
```diff
@@ -0,0 +1,91 @@
...
+};
+
+const useProgress = (service: string, options?: QueryHookOptions): Progress => {
```
Comment (dgdavid) [inline web/src/queries/progress.ts:47]: NP: `QueryHookOptions`? Do you mean `useProgressOptions`?

**Review by dgdavid — APPROVED** [review]
LGTM, just minor comments

File: web/src/queries/progress.ts (hunk 2 above)
Comment (imobachgs) [reply web/src/queries/progress.ts:47]: No, I meant `QueryHookOptions`... `UseProgressOptions` is a left-over. Good catch!

File: web/src/queries/progress.ts (hunk 1 above)
Comment (imobachgs) [reply web/src/queries/progress.ts:44]: It is a left-over. I will remove it.

**Review by dgdavid — APPROVED** [review]
Thanks for documenting the queries. It looks good, you can merge it.

**Pull request merged**